hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/AbstractDomainPath.java

```diff
@@ -252,7 +252,7 @@ private void addSortSpecification(
 			SortDirection sortOrder,
 			NullPrecedence nullPrecedence,
 			SqlAstCreationState creationState) {
-		final TableReference tableReference = tableGroup.resolveTableReference( getNavigablePath(), selection.getContainingTableExpression() );
+		final TableReference tableReference = tableGroup.resolveTableReference( null, selection.getContainingTableExpression() );
```

Comment:
The `resolve(...)` method at the top also uses `getNavigablePath()` to resolve the table reference. Wouldn't that also be wrong then?

Reply:
Applied the same logic to the `resolve()` method.